### PR TITLE
remove operators from word and add sub- and superscript captures

### PR DESCRIPTION
**Pull request by ryleelyman**
I prevented the characters "_", "+", "/", "-", "^" and "*" from being captured as part of a word, allowing them to be properly highlighted as operators even in constructs like `$n+m$` without spaces. This has the perhaps unwelcome side-effect of hyphens being highlighted as if they were minus signs.

The bigger change is the addition of `superscript` and `subscript` nodes; I found I wasn't able to properly [conceal them](https://github.com/ryleelyman/latex.nvim) without a dedicated node.

### CODE REVIEW

**Comment by dwuggh** [comment]
> This has the perhaps unwelcome side-effect of hyphens being highlighted as if they were minus signs.

perhaps we can disable it in `\text{}` ? It's marked as `text_mode` in the syntax tree.

File: grammar.js [hunk 1]
```diff
@@ -572,7 +596,7 @@ module.exports = grammar({
         )
       ),
 
-    command_name: $ => /\\([^\r\n]|[@a-zA-Z:_]+\*?)?/,
+    command_name: $ => /\\([^\r\n]|[@a-zA-Z]+\*?)?/,
```
Comment (pfoerster) [inline grammar.js:599]: Removing `:` and `_` here breaks recognizing `LaTeX3` commands (https://mirror.dogado.de/tex-archive/macros/latex/contrib/l3kernel/expl3.pdf) but I think we should add a dedicated rule for `LaTeX3` commands instead, which follows the syntax defined in the PDF more closely.

**Review by pfoerster — COMMENTED** [review]
Looks good to me.
The `LaTeX3` commands should probably be fixed in another PR. Currently, they are kinda broken too. 

@ryleelyman Can you fix the tests (`tree-sitter test -U`), please?
Thanks for contributing!

**Comment by pfoerster** [comment]
> That's certainly something that could be done, but I'm not sure whether it's worth the added complexity.

Yeah, this could definitely be done but there are a lot more cases to consider like specific math environments (`equation`, `align`, ...) so I probably would keep it simple for now.

> the idea is that the treesitter grammar has a node type it recognizes called "word," and currently that node type is the same regardless of whether or not the word is inside of math mode

The `word` rule is mostly there for historic reasons; the grammar itself was split off from [`texlab`](https://github.com/latex-lsp/texlab) and `texlab` mostly does not care about superscripts/subscripts and the distinction between math mode and text mode.

File: grammar.js (hunk 1 above)
Comment (ryleelyman) [reply grammar.js:599]: ahh, this is the first time I'm learning about `Latex3` haha. I'd be happy to work on this if you'd like.

**Comment by ryleelyman** [comment]
Tests should now be fixed, and I added a new one with superscripts and subscripts.

File: grammar.js (hunk 1 above)
Comment (pfoerster) [reply grammar.js:599]: > I'd be happy to work on this if you'd like.

Sure :+1:

**Comment by pfoerster** [comment]
Thanks!

**Pull request merged**